weblab/entities/forms.py

```diff
@@ -32,6 +37,30 @@ def entity_type(self):
         return self._meta.model.entity_type
 
 
+class EntityRenameForm(UserKwargModelFormMixin, forms.ModelForm):
+    """Used for renaming an existing entity."""
+
+    def clean_name(self):
+        name = self.cleaned_data['name']
+        if self._meta.model.objects.filter(author=self.user, name=name).exists():
```

Comment:
This looks like what I'd expect... so not sure why it isn't working! Maybe put some print calls in to see whether it's actually being used, and what the DB contains here if it is?

Reply:
thanks Jonathan, it appears to run this line fine.

Reply:
in the database are the two models my model1, test model 2 so still not clear why getting the database constraint issue

Reply:
It looks like `self.user` is `None` in the form, so the filter always returns false. The patch below resolves the issue; I can't commit it locally because of flake8 issues elsewhere! Basically you need to add `UserFormKwargsMixin` as a superclass for the view so it passes the user through to the form.

```diff
diff --git c/weblab/entities/tests/test_views.py i/weblab/entities/tests/test_views.py
index b1d0c81..31042b4 100644
--- c/weblab/entities/tests/test_views.py
+++ i/weblab/entities/tests/test_views.py
@@ -146,9 +146,9 @@ class TestEntityRenaming:
             data={
                      'name': 'test model 2'
                  })
-        assert response.status_code == 302
+        assert response.status_code == 200
         entity = ModelEntity.objects.first()
-        assert entity.name == 'test model 2'
+        assert entity.name == 'my model1'

     def test_model_and_protocol_renaming_success(self, client, logged_in_user, helpers):
         helpers.add_permission(logged_in_user, 'create_model')
@@ -186,7 +186,7 @@ class TestEntityRenaming:
 @pytest.mark.django_db
 @pytest.mark.parametrize("recipe,url,list_url", [
     (recipes.model, '/entities/models/%d/delete', '/entities/models/'),
-    (recipes.protocol, '/entities/protocols/%d/delete', '/entit1ies/protocols/'),
+    (recipes.protocol, '/entities/protocols/%d/delete', '/entities/protocols/'),
 ])
 class TestEntityDeletion:
     def test_owner_can_delete_entity(
diff --git c/weblab/entities/views.py i/weblab/entities/views.py
index d93fa89..c87005c 100644
--- c/weblab/entities/views.py
+++ i/weblab/entities/views.py
@@ -779,7 +779,7 @@ class EntityFileDownloadView(EntityTypeMixin, EntityVersionMixin, SingleObjectMi
         return response


-class RenameView(LoginRequiredMixin, UserPassesTestMixin, FormMixin, EntityTypeMixin, DetailView):
+class RenameView(LoginRequiredMixin, UserPassesTestMixin, EntityTypeMixin, UserFormKwargsMixin, FormMixin, DetailView):
     template_name = 'entities/entity_rename_form.html'
     context_object_name = 'entity'

```